### DISCUSSION TITLE
401-es válasz

### DISCUSSION THREAD

**Discussion by rrd108** [nav-gov-hu/M2M#46] [Kérdések] [answered]
Helló

Ma álltam neki megnézni, hogy hogyan is működik az API és az elején elakadtam.

A dev a `NavM2mCommon/tokenService/Token` hívásra mindig 401-et kapok saját fejlesztői gépről, de még a swaggerből is. 

Szóval gondolom valamit kihagyok, de nem találom, hogy mit.

**Comment by TamasPill** [comment, marked as answer]
A swaggerhub-on az alábbi server cím + végpont path alapján lehet tokent kérni:
https://m2m-dev.nav.gov.hu/rest-api/1.1/NavM2mCommon/tokenService/Token

**Reply by rrd108** [reply]
köszi. tényleg az url nem stimmelt

**Comment by GaborKiss82** [comment]
Sziasztok!

Nem akartam új bejegyzést, a 401-es kódra.
Valamiért a 
https://m2m-dev.nav.gov.hu/rest-api/1.1/NavM2mCommon/userregistrationService/Nonce
végpont nekem szintén HTTP ERROR 401 -et ad vissza. Teszt és éles végpontokon is, a swaggerhub-on és a kliensemen is. Van másnak is hasonló tapasztalata? 

curl -X 'POST' \
  'https://m2m-dev.nav.gov.hu/rest-api/1.1/NavM2mCommon/userregistrationService/Nonce' \
  -H 'accept: application/json' \
  -H 'messageId: 1a833dbd-2719-44e5-abeb-b6ffecc91a9a' \
  -H 'Content-Type: application/json' \
  -d '{
  "requestData": {
    "nonce": "XXXXXXXXXX"
  }
}'

**Reply by TamasPill** [reply]
A nonce beváltás előtt is kell már tokent kérni és a request header-ben elhelyezni (Authorization)

**Reply by GaborKiss82** [reply]
Köszönöm a segítséget, valóban ez hiányzott.

**Reply by e-lias** [reply]
És azt ezzel a végponttal és hozzá leírt paraméterekkel kell? /NavM2mCommon/tokenService/Token

**Reply by TamasPill** [reply]
Igen. Érdemes használni a Postman tesztcsomagot, ott egyből lehet próbálgatni is a saját paraméterekkel.
